backend/es/management/commands/index.py

```diff
@@ -16,60 +16,72 @@ class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument(
             'corpus',
-            help='''Sets which corpus should be indexed. Use the name provided in settings.py''',
+            help='''Sets which corpus should be indexed. This should match the "name"
+                field in the database. For Python corpora, this field is based on the
+                name in settings.py''',
         )
 
         parser.add_argument(
             '--start', '-s',
-            help='''The date where indexing should start.
-            The input format is YYYY-MM-DD.
-            If not set, indexing will start from corpus minimum date.'''
+            help='''Minimum date to select documents. The input format is YYYY-MM-DD.
+            Optional. Only has effect for Python corpora which implement date selection
+            in their sources() method. No effect in combination with --mappings-only.'''
         )
 
         parser.add_argument(
             '--end', '-e',
-            help='''The date where indexing should end
-            The input format is YYYY-MM-DD.
-            If not set, indexing will stop at corpus maximum date.'''
+            help='''Maximum date to select documents. The input format is YYYY-MM-DD.
+            Optional. Only has effect for Python corpora which implement date selection
+            in their sources() method. No effect in combination with --mappings-only.'''
         )
 
         parser.add_argument(
             '--delete', '-d',
             action='store_true',
-            help='Delete the index before indexing'
+            help='''If this job is set to create an index that already exists, delete
+                it instead of raising an exception. No effect in combination with
+                --add.'''
```

Comment:
I think in case arguments contradict eachother, it's better to cancel the runtime and give an error. Silently performing either `add` or `delete` may lead to unexpected results, even when documented in the help prompt.